### DISCUSSION TITLE
Spotify Premium Credentials

### DISCUSSION THREAD

**Discussion by shardsofaperture** [dupontgu/retro-ipod-spotify-client#51] [Q&A]
So I am still stuck at the point where my spotipod boots up to the screen with no cure and then just stays there. Though vnc I can move the mouse, get a menu, go to desktop (I'm using the gui instead of command line only raspbian because why not.

I'm wondering, when I went though and made all my tokens, app id's, Are made with my NON PREMIUM ��account. Should this make a difference? Also if I try to log in with a non premium account will that break the bootup? 

I have also attempted to use a premium login which was loaned to me to test however same issue, I only get openbox with a black screen.

I still think I'm having issues with the cached stuff as well, I managed to do that url to generate something once (the one from issue#7) but since that one time a month ago no luck.

Any thoughts are welcomed!

**Comment by tomaculum** [comment]
There are some parts of the Spotify API, where you need a premium account but I think this only applies to the web connect API, which controls the playback of other devices. Since the issue you are facing is still there after using a premium account, I would assume that there's another issue somewhere.

Do you see a cursor on the black screen? Or something else which would indicate that the screen does work?

**Reply by shardsofaperture** [reply]
To clarify, th login name/password was premium.  The dev API keys were made using a free account. I don't think that should make a difference but I don't know.

Regarding screen, I have the black no cursor screen with openbox that starts up but it doesn't launch the spotify frontend. I can get a context menu in openbox, exit it, but that's it, spod stuff doesn't auto launch. 

 I've checked over and over that all my keys are right, the token .cache is in the frontend folder, ouath demo shows its logging in with the correct premium account. Spotifypod manager like 143 or whatever is set to the correct name, I build click.c, 

I know there is a way to manually launch the program though terminal using pip3 I think, my guess is something is causing it to crash before it loads (which is why I was wondering about the keys being from a non premium acct)

**Reply by tomaculum** [reply]
Yes, I assume the login credentials are the relevant part and not the api key part. 

Could you start any other gui Programm and check if the openbox part is configured correctly? Maybe some kind of tkinter hello world?

**Reply by shardsofaperture** [reply]
HA ok so I created a hello world tkinter script and found out I was getting a missing module error. Turns out running

` pip install tkintertable`

fixed whatever was wrong with my tkinter. 

I then commented everything out in autostart to show just 

`python helloworld.py`

And hello world DOES work  at bootup. So I at least know I'm good there.

I rebooted and it still did not work. I went back though the issues, and found on #41 that adding the env variables to the bash profile might help. 

`sudo nano ~/.bash_profile`

so I did that and replaced 

`sudo -H -u pi python3 spotifypod.py &` 

with

 `sudo -H -u pi --preserve-env=SPOTIPY_REDIRECT_URI,SPOTIPY_CLIENT_ID,SPOTIPY_CLIENT_SECRET python3 spotifypod.py &`

and then rebooted again.

Open box is still black. I'm really at a loss what I could have done wrong and just about ready to start over AGAIN.

I will say if i just run from the frontend dir 

`sudo -H -u pi --preserve-env=SPOTIPY_REDIRECT_URI,SPOTIPY_CLIENT_ID,SPOTIPY_CLIENT_SECRET python3 spotifypod.py `

I get the following `raceback (most recent call last):
  File "spotifypod.py", line 13, in <module>
    from view_model import *
  File "/home/pi/rsip/frontend/view_model.py", line 1, in <module>
    import spotify_manager
  File "/home/pi/rsip/frontend/spotify_manager.py", line 82, in <module>
    sp = spotipy.Spotify(auth_manager=SpotifyOAuth(scope=scope))
  File "/home/pi/.local/lib/python3.7/site-packages/spotipy/oauth2.py", line 313, in __init__
    self.client_id = client_id
  File "/home/pi/.local/lib/python3.7/site-packages/spotipy/oauth2.py", line 94, in client_id
    self._client_id = _ensure_value(val, "client_id")
  File "/home/pi/.local/lib/python3.7/site-packages/spotipy/oauth2.py", line 70, in _ensure_value
    raise SpotifyOauthError(msg)
spotipy.oauth2.SpotifyOauthError: No client_id. Pass it or set a SPOTIPY_CLIENT_ID environment variable`

I confirmed the .cache file is in the frontend dir, I know the IDs are all set up.... I have NO idea where to go from here. I would think running it directly would work but its not.

**Reply by tomaculum** [reply]
Great! Since the hello world example did work well, maybe try to replace the "hello world" text with the client_id environment variable? This way you can see if the environment variable setup does work. 

It is suspicious that the error mentions explicitly that the client_id environment variable is missing. If there would be an issue with your account or something with the code inside this repository, the error message would be different.

**Comment by shardsofaperture** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}                                                                                                                                         No I haven't got back to this.... project is sitting in a drawer lol      Sent via BlackBerry Hub+ Inbox for Android                                                                                                                                                                           From: ***@***.***: August 15, 2022 18:46To: ***@***.***: ***@***.***: ***@***.***; ***@***.***: Re: [dupontgu/retro-ipod-spotify-client] Spotify Premium Credentials (Discussion #51)  
Has your problem been corrected?

**Comment by shardsofaperture** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Thank you, hopefully I can find time over the next week to check this out!                                                                                                                                             Sent via BlackBerry Hub+ Inbox for Android                                                                                                                                                                           From: ***@***.***: August 15, 2022 20:25To: ***@***.***: ***@***.***: ***@***.***; ***@***.***: Re: [dupontgu/retro-ipod-spotify-client] Spotify Premium Credentials (Discussion #51)  
let me know if the above gets you any further.

**Reply by doris1347** [reply]
See issue #69!
